core/Segment/SegmentExpression.php

```diff
@@ -253,21 +257,38 @@ protected function getSqlMatchFromDefinition($def, &$availableTables)
         // We match NULL values when rows are excluded only when we are not doing a
         $alsoMatchNULLValues = $alsoMatchNULLValues && !empty($value);
 
-        if ($matchType === self::MATCH_ACTIONS_CONTAINS
-            || is_null($value)
-        ) {
-            $sqlExpression = "( $field $sqlMatch )";
-        } else {
-            if ($alsoMatchNULLValues) {
-                $sqlExpression = "( $field IS NULL OR $field $sqlMatch ? )";
+        $sqlExpressions = array();
+        $values = array();
+        foreach ($fields as $field) {
+            $sqlMatchReplaced = str_replace('%s', $field, $sqlMatch);
```

Comment:
Can you check / make sure that no SQL injection is possible here? Not sure where `$fields` or `$def[0]` is coming from and whether they were validated against a whitelist or so.